unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2955,14 +2986,16 @@ abstract public class BaseFPSAgentController {
                                 }
                             }
 
-                            // if we see at least one vis point, the object is "visible"
-                            if (visPointCount > 0) {
 #if UNITY_EDITOR
-                                sop.debugIsVisible = true;
-#endif
-                                if (!currentlyVisibleItems.Contains(sop)) {
-                                    currentlyVisibleItems.Add(sop);
-                                }
+                            sop.debugIsVisible = visCheck.visible;
+                            sop.debugIsInteractable = visCheck.interactable;
+#endif                
+                            if (visCheck.visible && !currentlyVisibleItems.Contains(sop)) {
+                                currentlyVisibleItems.Add(sop);
+                            }
+
+                            if (visCheck.interactable && !interactableItems.Contains(sop)) {
+                                interactableItems.Add(sop);
```

Comment:
It seems like doing these `Contains` checks might be slow as `currentlyVisibleItems` and `interactableItems` are lists. Why not make these `HashSets`?

Reply:
these have been updated to use HashSets instead of lists